engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditionChecker.java

```diff
@@ -97,6 +97,10 @@ public final class ProcessInstanceMigrationPreconditionChecker {
       "Expected to migrate process instance '%s' but a concurrent command was executed on the process instance. Please retry the migration.";
   private static final long NO_PARENT = -1L;
 
+  /**
+   * Exception that can be thrown during the migration of a process instance, in case no process
+   * instance with the given key exists.
+   */
   public static void requireNonNullProcessInstance(
```

Comment:
❌ This method throws an exception when record is null, not specifically if there is no process instance with the given key.

This JavaDoc and that of the other methods describe when these methods should be used rather than what they do and how they should be used. IMO JavaDoc, should clearly describe what a method does and how it can be used, i.e. what should be provided as `record` parameter and what should be provided as `processInstanceKey` + what are those parameters used for.